src/backend/common/helpers/event_team_status_helper.py

```diff
@@ -432,7 +432,11 @@ def _build_alliance_info(
             return None
 
         # Calculate the role played by the team on the alliance
-        backup_info = alliance.get("backup", {}) if alliance.get("backup") else {}
+        backup_info = (
+            cast(EventAllianceBackup, alliance["backup"])
+            if alliance.get("backup")
+            else {}
+        )
```

Comment:
cc @phil-lopreiato This file changed in two places due to the changes in https://github.com/the-blue-alliance/the-blue-alliance/pull/3432/files#diff-5c40aafca78f0698e6d073af075cb8120c7a4cdf3cea4615406def4fe52fab67 - could you confirm this looks right?

Also - why can't we do `alliance.backup`? Do the typed dicts not give us shorthand semantics? I tried adding a property to the `TypedDict` class without any luck.

Reply:
I'm not sure why we need the cast here? Is this because of the default access thing?

Also, I don't think dot notation works for python dicts?